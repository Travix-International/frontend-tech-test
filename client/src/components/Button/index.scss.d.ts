export const btn: string;
export const normal: string;
export const primary: string;
export const icon: string;
