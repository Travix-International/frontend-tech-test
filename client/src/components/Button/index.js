import "./styles.scss";
import React from "react";

const Button = ({ className, primary, children, ...rest }) => {
  const classNames = `button ${primary ? "button_primary" : ""} ${className}`;

  return (
    <button className={classNames} {...rest}>
      {children}
    </button>
  );
};

export default Button;
