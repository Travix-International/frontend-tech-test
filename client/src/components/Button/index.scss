@import '../../styles/theme.scss';

.btn {
    > i,
    > span {
        display: inline-block;
        pointer-events: none;
    }
    font-size: 1rem;
    line-height: 2rem;
    border-radius: 10px;
    padding: .2rem 1rem;
    font-weight: bold;
    transition: all .3s;
    outline: none;

    &:hover {
        cursor: pointer;
    }
}

.normal {
    color: $color-text;
    background-color: transparent;
    border: 1px solid $color-primary;
}

.primary {
    color: #fff;
    background-color: $color-primary;
    border: 1px solid $color-primary;

    &:hover {
        background-color: lighten($color-primary, 10%);
        border: 1px solid lighten($color-primary, 10%);
    }
}

.icon {
    border-radius: none;
    border: none;
    background-color: transparent;
    color: lighten($color-accent, 20%);
    &:hover{
        color: $color-accent;
    }
}