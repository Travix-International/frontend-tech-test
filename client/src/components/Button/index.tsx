import React from 'react';
import styles from './index.scss';

type ButtonType = 'primary' | 'normal' | 'icon';

interface Props {
    type?: ButtonType;
    title?: string;
    onClick?: () => void;
}

const Button: React.FC<Props> = ({ type = 'normal', title, onClick, children }) => {
    const handleClick = (e: React.MouseEvent<HTMLElement>) => {
        if (onClick) { onClick(); }
    };

    return (
        <button title={title} className={`${styles.btn} ${styles[type as ButtonType]}`} onClick={handleClick}>{children}</button>
    );
};

export default Button;