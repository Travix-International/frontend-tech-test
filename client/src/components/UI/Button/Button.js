import React from 'react'
import PropTypes from 'prop-types'

import styles from './Button.scss'

const Button = props => {
  return (
    <button
      className={`${props.add ? styles.Button_Add : null} ${
        props.delete ? styles.Button_Delete : null
      }`}
    >
      {props.text}
    </button>
  )
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  add: PropTypes.bool,
  delete: PropTypes.bool
}

export default Button
