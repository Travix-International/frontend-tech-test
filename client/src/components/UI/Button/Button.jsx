import React from 'react';

import styles from './Button.module.scss';

const Button = props => (
  <button {...props} className={styles.Button}>{props.children}</button>
);

export default Button;