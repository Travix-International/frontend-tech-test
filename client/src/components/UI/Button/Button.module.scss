.Button {
  font-family: 'Khand';
  height: 45px;
  line-height: 41px;
  margin-bottom: 3px;
  padding: 0 24px 0 27px;
  font-size: 22px;
  background-color: #005fbb;
  border-radius: 5px;
  box-shadow: 0 3px #45619d, -5px 8px rgba(0, 0, 0, .15);
  color: #fff;
  cursor: pointer;
}