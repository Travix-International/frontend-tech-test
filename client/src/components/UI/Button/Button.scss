@import '../../../scss/vars.scss';
.Button {
  position: relative;
  overflow: hidden;
  margin: 1rem;
  padding: 1rem;
  font-size: 1rem;
  text-transform: uppercase;
  border-radius: 1rem;
  border: none;
  color: white;
  transition: all 0.5s ease-in-out;
  outline: none;
  &:active:after {
    padding: 0;
    margin: 0;
    opacity: 1;
    transition: 0s
  }
  &:after {
    content: "";
    display: block;
    position: absolute;
    border-radius: 50%;
    padding-top: 240%;
    padding-left: 240%;
    margin-left: -120%;
    margin-top: -120%;
    opacity: 0;
    transition: all 0.8s
  }
  &_Add {
    @extend .Button;
    background-color: $darkgreen;
    &:hover {
      background-color: lighten($darkgreen, 15%);
      font-size: 2rem;
      transition: all 0.5s ease-in-out;
      cursor: pointer;
    }
    &:after {
      background-color: darken($darkgreen, 15%)
    }
  }
  &_Delete {
    @extend .Button;
    background-color: $red;
    &:hover {
      background-color: lighten($red, 15%);
      font-size: 2rem;
      transition: all 0.5s ease-in-out;
      cursor: pointer;
    }
    &:after {
      background-color: darken($red, 15%)
    }
  }
}