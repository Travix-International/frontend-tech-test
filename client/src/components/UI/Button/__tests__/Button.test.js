import React from 'react'
import {render, cleanup} from 'react-testing-library'
import Button from '../Button'

describe('App should render properly', () => {
  afterEach(cleanup)

  it('Should match snapshot', () => {
    const {container} = render(<Button text={'add'} />)
    expect(container.firstChild).toMatchSnapshot('Button_snapshot_1')
  })

  it('Should show "add button" if passed text=add', () => {
    const {getByText} = render(<Button text={'add'} />)
    getByText('add')
  })
  it('Should show "delete button" if passed text=delete', () => {
    const {getByText} = render(<Button text={'delete'} />)
    getByText('delete')
  })
  it('Should show "edit button" if passed text=edit', () => {
    const {getByText} = render(<Button text={'edit'} />)
    getByText('edit')
  })
})
