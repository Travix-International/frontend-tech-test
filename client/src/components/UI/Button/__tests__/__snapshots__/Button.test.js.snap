// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly Should match snapshot: Button_snapshot_1 1`] = `
<button
  class="null null null"
>
  add
</button>
`;
